sdk/storage/Azure.Storage.Blobs.Cryptography/src/EncryptedBlobClient.cs

```diff
@@ -0,0 +1,330 @@
+    /// </summary>
+    public class EncryptedBlobClient : BlobClient
+    {
+
+        /// <summary>
+        /// The wrapper used to wrap the content encryption key.
+        /// </summary>
+        private IKeyEncryptionKey KeyWrapper { get; }
+
+        /// <summary>
+        /// The algorithm identifier to use with <see cref="KeyWrapper"/>.
+        /// </summary>
+        private string KeyWrapAlgorithm { get; }
+
+        #region ctors
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EncryptedBlobClient"/>
+        /// class for mocking.
+        /// </summary>
+        protected EncryptedBlobClient()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EncryptedBlobClient"/>
+        /// class.
+        /// </summary>
+        /// <param name="connectionString">
+        /// A connection string includes the authentication information
+        /// required for your application to access data in an Azure Storage
+        /// account at runtime.
+        ///
+        /// For more information, <see href="https://docs.microsoft.com/en-us/azure/storage/common/storage-configure-connection-string"/>.
+        /// </param>
+        /// <param name="containerName">
+        /// The name of the container containing this encrypted block blob.
+        /// </param>
+        /// <param name="blobName">
+        /// The name of this encrypted block blob.
+        /// </param>
+        /// <param name="keyEncryptionKey">
+        /// Required for uploads. Provider to wrap the one-time content encryption key via the envelope technique.
+        /// </param>
+        /// <param name="keyResolver">
+        /// Required for downloads. Provider to get the correct <see cref="IKeyEncryptionKey"/> for a given download.
+        /// The fetched <see cref="IKeyEncryptionKey"/> will be used to unwrap the one-time content encryption key via
+        /// the envelope technique.
+        /// </param>
+        /// <param name="encryptionKeyWrapAlgorithm">
+        /// Required for uploads. String identifier of the key wrapping algorithm the client should use with
+        /// <paramref name="keyEncryptionKey"/>.
+        /// </param>
+        /// <param name="options">
+        /// Optional client options that define the transport pipeline
+        /// policies for authentication, retries, etc., that are applied to
+        /// every request.
+        /// </param>
+        public EncryptedBlobClient(
+            string connectionString,
+            string containerName,
+            string blobName,
+            IKeyEncryptionKey keyEncryptionKey = default,
+            IKeyEncryptionKeyResolver keyResolver = default,
+            string encryptionKeyWrapAlgorithm = default,
+            BlobClientOptions options = default)
+            : base(connectionString, containerName, blobName, FluentAddPolicy(options, new ClientSideDecryptionPolicy(keyResolver, keyEncryptionKey)))
```

Comment:
NIT - Maybe consider making a new line for each parameter for this? If not I'm okay with it